doc/15-troubleshooting.md

```diff
@@ -317,6 +317,111 @@ not `assign where match("MySQL*", host.name) && host.vars.production`.
 For all test hosts `host.vars.production` will be false and terminate
 the `&&` rather than also evaluating `match("MySQL*", host.name)`.
 
+### Try reducing concurrency (threads)
+
+Yes, reducing and not increasing. By default, Icinga 2 already starts as many
+threads as there are CPU cores according to the OS (unaware of SMT aka
+HyperThreading). So there's no point in increasing as Icinga would not gain
+additional CPU time.
+
+But more threads also require more synchronization between them. This may
+outweigh the CPU time gained by starting X threads instead of just one. (In case
+of more than 12 cores/threads, see proof of concept below.) So reducing may
+indeed help or at least save CPU time and power at no cost.
+
+Start with benchmarking your Icinga 2 config with `time icinga2 daemon -C` on
+the node in question. The results will be most accurate during normal operation,
+i.e. while Icinga is running, but not reloading (e.g. due to config deployments).
+
+Icinga accepts the argument `-DConfiguration.Concurrency=` with the number (of
+threads) immediately after the "=". Start with one and finish with the number of
+CPU cores. Write down the times. I.e.:
+
+* `time icinga2 daemon -C -DConfiguration.Concurrency=1`
+* `time icinga2 daemon -C -DConfiguration.Concurrency=2`
+* `time icinga2 daemon -C -DConfiguration.Concurrency=3`
+* ...
+
+If significantly less threads than CPU cores significantly reduce the time
+(reported as "real") Icinga 2 needs to load its configuration, pick the number
+with the best time and persist it in your init daemon. In case of systemd locate
+a file named "icinga2.service" and copy the `ExecStart=` line. Next, run
+`systemctl edit icinga2.service`. This will open an editor. Add `[Service]` (if
+not already present) and the copied `ExecStart=` line. Append
+`-DConfiguration.Concurrency=` and the chosen number so that the result looks
+like this:
+
+```
+[Service]
+ExecStart=/usr/sbin/icinga2 daemon --close-stdio -e ${ICINGA2_ERROR_LOG} -DConfiguration.Concurrency=42
+```
+
+Save the file and close the editor. Restart Icinga.
+Finally verify whether your changes took effect and enjoy the speed.
+
+#### Proof of concept
+
+* Processor: "Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz"
+* Cores according to OS: 56
+* RAM: 256 GB
+* Swap: none
+* Virtualization: Podman
+* Icinga: v2.14.0
```

Comment:
I still think most of the information is irrelevant here.
- RAM/swap: as long as there's enough of it for the process and some filesystem cache, the exact number doesn't really have an effect on the result.
- Virtualization: well it's Linux namespaces, that's cheap, you wouldn't get different numbers from running it directly on the host system (or if you do, it's more likely due to different compilers used or something like that).

If the section was boiled down to something like "Icinga 2.14.0 on a 56 core machine (28 physical cores, 58 hardware threads)", I don't think any relevant information would be lost.

Reply:
> * RAM/swap: as long as there's enough of it for the process and some filesystem cache, the exact number doesn't really have an effect on the result.

I'm afraid, "enough" is a subjective term on its own. The number's point is to prove (this is a _proof_ of concept after all) that it's more than "enough". In combination with the swap amount it proves that the system never wastes measured time for swapping.

> If the section was boiled down to something like "Icinga 2.14.0 on a 56 core machine (28 physical cores, 58 hardware threads)", I don't think any relevant information would be lost.

This is an empirical analysis, not some claim from someone on the internet.